File: Vibechat.Web/Vibechat.Web/ClientApp/src/app/Data/Message.ts
import {AppUser} from "./AppUser";
import {Attachment} from "./Attachment";
import {MessageType} from "./MessageType";
import {ChatEvent} from "./ChatEvent";

export class Message {
  constructor(init?: Partial<Message>) {
    (<any>Object).assign(this, init);
  }

  public id: number;

  public user: AppUser;

  public messageContent: string;

  public conversationID: number;

  public state: number;

  public timeReceived: Date | string;

  public attachmentInfo: Attachment;

  public forwardedMessage: Message;

  public type: MessageType;

  public event: ChatEvent;

  public encryptedPayload: string;
}

File: Vibechat.Web/Vibechat.Web.Data/DTO/Messages/Message.cs
using Vibechat.Web.ChatData.Messages;
using Vibechat.Web.DTO.Messages;

namespace VibeChat.Web.ChatData
{
    public class Message
    {
        public int Id { get; set; }
        public AppUserDto User { get; set; }
        public string MessageContent { get; set; }
        public int ConversationID { get; set; }
        public string TimeReceived { get; set; }

        public Message ForwardedMessage { get; set; }
        
        public MessageType Type { get; set; }
        
        public ChatEvent Event { get; set; }

        public MessageAttachment AttachmentInfo { get; set; }

        public MessageState State { get; set; }

        /// <summary>
        ///     if this message is from private chat, store payload there.
        /// </summary>
        public string EncryptedPayload { get; set; }
    }
}